keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ApplicationEvent.kt

```diff
@@ -0,0 +1,61 @@
+package com.netflix.spinnaker.keel.events
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import java.time.Clock
+import java.time.Instant
+
+@JsonTypeInfo(
+  use = JsonTypeInfo.Id.NAME,
+  property = "type",
+  include = JsonTypeInfo.As.PROPERTY
+)
+@JsonSubTypes(
+  JsonSubTypes.Type(value = ApplicationActuationPaused::class, name = "ApplicationActuationPaused"),
+  JsonSubTypes.Type(value = ApplicationActuationResumed::class, name = "ApplicationActuationResumed")
+)
+sealed class ApplicationEvent : PersistentEvent() {
```

Comment:
One thing that occurs to me is that this name will conflict with a pretty common Spring class. I don't think we use it so it's not a huge deal, but we might want to reconsider.

Reply:
Actually it did bit me already. `KeelApplicationEvent`?